src/main/java/goodgenerator/blocks/tileEntity/PreciseAssembler.java

```diff
@@ -212,8 +216,8 @@ public boolean checkRecipe_EM(ItemStack itemStack) {
         }
         else {
             for (GT_MetaTileEntity_Hatch_InputBus bus : mInputBusses) {
-                if (!isValidMetaTileEntity(bus)) continue;
-                GT_Recipe tRecipe = getRecipeMap().findRecipe(this.getBaseMetaTileEntity(), true, Math.min(getMachineVoltageLimit(), getMaxInputVoltage()), inputFluids, getStoredItemFromHatch(bus));
+                if (!isValidMetaTileEntity(bus) && getStoredItemFromHatch(bus).length < 1) continue;
```

Comment:
Wrong logic, should be like that
```suggestion
                if (!isValidMetaTileEntity(bus) || getStoredItemFromHatch(bus).length < 1) continue;
```